### PR TITLE
Docs: fixing wrong jinja example and stating how airbyte decides last date cursor

### PR DESCRIPTION
**Pull request by Swadish**
## What
*Documents how Airbyte decides which date do pick as the state to use in the next sync
*Fixes an example that was using {{stream_interval.start_date}} and {{ stream_interval.end_date }} instead of the correct way {{ stream_interval.start_time }} and {{ stream_interval.end_time }}

## How
*Improving docs trying to state how Airbyte decides which date to store in state, currently the docs only says that it uses the last record "updated_at" instead of saying that it compares the last record "updated_at" to the current time that the sync is being run.*

## 🚨 User Impact 🚨
*No breaking changes, just trying to save some hours for the next person that tries to understand how airbyte decides the state to be persisted.*

Feel free to suggest other wording on this PR.

### CODE REVIEW

**Comment by natikgadzhi** [comment]
@Swadish, thank you for the pull request! I'm just onboarding and can take a look later today if the change is correct, and we can merge it up after the code freeze (Jan 2)

File: docs/connector-development/connector-builder-ui/incremental-sync.md [hunk 1]
```diff
@@ -82,10 +82,12 @@ Then when a sync is triggered for the same connection the next day, the followin
 curl 'https://content.guardianapis.com/search?from-date=<b>2023-04-15T07:30:58Z</b>&to-date={`<now>`}'
 </pre>
 
+Except if the last record found has a date less than the time that the sync was run. In this case the sync run time will be stored in the state.
```
Comment (girarda) [inline docs/connector-development/connector-builder-ui/incremental-sync.md:85]:
```suggestion
:::info
If the last record read has a datetime earlier than the end time of the stream interval, the end time of the interval will be stored in the state. In this case, the sync start time will be stored in the state
:::
```

File: docs/connector-development/connector-builder-ui/incremental-sync.md [hunk 2]
```diff
@@ -82,10 +82,12 @@ Then when a sync is triggered for the same connection the next day, the followin
...
+
 The `from-date` is set to the cutoff date of articles synced already and the `to-date` is set to the current date.
 
 :::info
-In some cases, it's helpful to reference the start and end date of the interval that's currently synced, for example if it needs to be injected into the URL path of the current stream. In these cases it can be referenced using the `{{ stream_interval.start_date }}` and `{{ stream_interval.end_date }}` [placeholders](/connector-development/config-based/understanding-the-yaml-file/reference#variables). Check out [the tutorial](./tutorial.mdx#adding-incremental-reads) for such a case.
+In some cases, it's helpful to reference the start and end date of the interval that's currently synced, for example if it needs to be injected into the URL path of the current stream. In these cases it can be referenced using the `{{ stream_interval.start_time }}` and `{{ stream_interval.end_time }}` [placeholders](/connector-development/config-based/understanding-the-yaml-file/reference#variables). Check out [the tutorial](./tutorial.mdx#adding-incremental-reads) for such a case.
```
Comment (girarda) [inline docs/connector-development/connector-builder-ui/incremental-sync.md:90]: good catch!

**Review by girarda — COMMENTED** [review]
Thanks for updating the docs! made a small suggestion

File: docs/connector-development/connector-builder-ui/incremental-sync.md (hunk 1 above)
Comment (Swadish) [reply docs/connector-development/connector-builder-ui/incremental-sync.md:85]: I think you forgot to delete the "In this case, the sync start time will be stored in the state"

But loved your suggestion, will commit it.

**Comment by girarda** [comment]
Created a non-forked PR that can be merged: https://github.com/airbytehq/airbyte/pull/34000

**Comment by girarda** [comment]
I cloned your PR and merged [it here](https://github.com/airbytehq/airbyte/pull/34000). Thank you for your contribution @Swadish !

**Pull request closed without merging**